mesonbuild/dependencies/base.py

```diff
@@ -1095,7 +1095,17 @@ def _get_cmake_info(self):
             return None
 
         # Extract the variables and sanity check them
+        root_paths = set(temp_parser.get_cmake_var('MESON_FIND_ROOT_PATH'))
+        root_paths.union(set(temp_parser.get_cmake_var('MESON_CMAKE_SYSROOT')))
+        root_paths = sorted(root_paths)
+        root_paths = list(filter(lambda x: os.path.isdir(x), root_paths))
         module_paths = sorted(set(temp_parser.get_cmake_var('MESON_PATHS_LIST')))
+        if len(root_paths) > 0:
+            unrooted_paths = module_paths
+            module_paths = set()
+            for j in root_paths:
+                for i in ['/'] + unrooted_paths:
+                    module_paths.add(j + i)
```

Comment:
You should use something platform-independent here (like `os.path.join`). You might also have to change the `'/'` in the for loop for this.

Reply:
How do you propose to remove `/` in `i` variable? `os.path.join()` ignores `j`, because `i` is absolute.

Reply:
First of all, what are you trying to do here? Are you trying to prepend the `root_paths` to the `module_paths`?

Reply:
Yes, I am trying to prepend `root_paths` to `module_paths`.

Reply:
Additionally, what are you trying to achieve with the `['/'] + unrooted_paths`? This will create a new list with `'/'` as its first element, followed by all elements in `unrooted_paths`, thus effectively adding the path in `j` to `module_paths`. If this is wanted, an empty string would also do the trick.

For the absolute path problem, you could have a look at this: https://stackoverflow.com/questions/50846049/join-two-absolute-paths

Yes. it's a lot more hacky, but it should work on all platforms.

Reply:
I use `Path()` and `Path.relative_to` to concatenate paths now.